lib/functions/single.js

```diff
@@ -0,0 +1,22 @@
+const { filter } = require("./filter");
+const { take } = require("./take");
+
+const single = (iterable, filterCondition = () => true) => {
+  const filtered = filter(iterable, filterCondition);
+  const filteredWithTake = take(filtered, 2);
+  const iterator = filteredWithTake[Symbol.iterator]();
+
+  const { done, value } = iterator.next();
+  if (done) {
+    throw new Error("There's no element!");
+  }
+
+  const nextObj = iterator.next();
```

Comment:
Why not use destructuring of the `{ done }` value? Give it a different name to avoid issues with the constant created before.